src/main/java/com/project/controller/CommentController.java

```diff
@@ -0,0 +1,77 @@
+package com.project.controller;
+
+import com.project.exception.*;
+import com.project.model.request.CommentRequest;
+import com.project.model.response.CommentResponse;
+import com.project.service.CommentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.validation.Valid;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+
+@RestController
+@CrossOrigin(origins = "*")
+@RequestMapping("api/applications/{appId}/comments")
+public class CommentController {
+    private final CommentService commentService;
+
+    @Autowired
+    public CommentController(CommentService commentService){
+        this.commentService = commentService;
+    }
+
+    @GetMapping
+    public ResponseEntity<List<CommentResponse>> fetchComments(@PathVariable("appId") Long appId) {
+        return new ResponseEntity<>(commentService.getAll(appId), HttpStatus.OK);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<CommentResponse> fetchComment(@PathVariable("id") Long id) throws CommentNotFoundException {
+        return new ResponseEntity<>(commentService.getById(id), HttpStatus.OK);
+    }
+
+    @GetMapping(value = "/{id}/attachment/{filename}")
```

Comment:
Why you chose `filename` as a path variable and not a file ID?

Reply:
well, as we can have only one file attachment per comment, it was convenient to make a field for it in the database rather than make a new table for files, which leaves files with no id. Another problem that I had was that if I didn't end the path with filename, it would not understand that it is the file that I uploaded and would download some nonsense file with no extension. I might have made a mistake somewhere while doing that, but when I changed to filename, it worked well, so I left it this way.